Flames/flames_drv.py

```diff
@@ -293,78 +295,125 @@ def checkSequence(self, firingSequence):
         except Exception as e:
             print("Error: firingSequence is malformed or out of bounds", e)
             traceback.print_exc(file=sys.stdout)
-            return False 
+            return False
+
+    ## send bangCommandList to the poofer controller boards
+    def firePoofers(self, bangCommandList):
+        #TODO: This was just grabbed from heartbeat_controller. Need to
+        # make sure this is what we want
+        try:
+            if not self.isRunning:
+                return 1
+
+            if not ser:
+                ser = initSerial()
 
+            for command in bangCommandList:
+                ser.write(command.encode())
 
-    ## parse out firing sequence into bang code ##
-    def generateBangCommandList(self, firingSequence):
+        except Exception as e:
+            ser.close()
+            ser = None
+            print("Error sending bangCommandSequence to poofer controller boards", e)
+            traceback.print_exc(file=sys.stdout)
+
+    def disablePoofer(msgObj):
+        self.disabled_poofers.add(msgObj["name"])
+        event_manager.postEvent({"msgType":"poofer_disabled", "id":msgObj["name"]})
+
+    def enablePoofer(msgObj):
+        enabled_poofer = self.disabled_poofers.pop(msgObj["name"])
+        if enabled_poofer != None:
+            event_manager.postEvent({"msgType":"poofer_enabled", "id":msgObj["name"]})
+
+    def resumeAll():
+        self.isStopped = False
+        event_manager.postEvent({"msgType":"global_resume", "id":"all?"})
+
+    def stopAll():
+        self.isStopped = True
+        # TODO: send bang command to stop all poofers
+        self.pooferEvents = list() # reset all pooferEvents
+        event_manager.postEvent({"msgType":"global_pause", "id":"all?"})
+
+    def startFlameEffect(msgObj):
+        if self.checkSequence(msgObj):
+            setUpEvent(msgObj)
+            event_manager.postEvent({"msgType":"sequence_start", "id":msgObj["name"]})
+
+    def stopFlameEffect(msgObj):
+        #  TODO: Need to figure out how to stop this,
+        # since now this is also removing commands to stop poofers
+        event_manager.postEvent({"msgType":"sequence_stop", "id":msgObj["name"]})
```

Comment:
I'd punt on stopping the poofers that are currently on. They don't stay on for more than 3 seconds anyway, so just removing upcoming poofer events should be fine.